### PR TITLE
feature: extract Base Message Class

### PR DESCRIPTION
**Pull request by LucasLi1024**

### CODE REVIEW

**Review by cyjseagull — APPROVED** [review]

File: bcos-boostssl/websocket/WsMessage.cpp [hunk 1]
```diff
@@ -39,30 +39,33 @@ bool WsMessage::encode(bytes& _buffer)
...
     _buffer.insert(_buffer.end(), (byte*)&status, (byte*)&status + 2);
-    _buffer.insert(_buffer.end(), m_seq->begin(), m_seq->end());
+    _buffer.insert(_buffer.end(), m_seq.begin(), m_seq.end());
     _buffer.insert(_buffer.end(), m_data->begin(), m_data->end());
 
     return true;
 }
 
-int64_t WsMessage::decode(const byte* _buffer, std::size_t _size)
+ssize_t WsMessage::decode(bytesConstRef _buffer)
```
Comment (ywy2090) [inline bcos-boostssl/websocket/WsMessage.cpp]: ssize_t unable compile on windows

File: bcos-boostssl/websocket/WsMessage.h [hunk 2]
```diff
@@ -46,30 +62,32 @@ class WsMessage
 public:
     WsMessage()
     {
-        m_seq = std::make_shared<bcos::bytes>(SEQ_LENGTH, 0);
+        std::string str(SEQ_LENGTH, '0');
+        m_seq = str;
```
Comment (ywy2090) [inline bcos-boostssl/websocket/WsMessage.h]: remove this

File: bcos-boostssl/websocket/WsMessage.h [hunk 3]
```diff
@@ -46,30 +62,32 @@ class WsMessage
...
-    virtual int64_t decode(const bcos::byte* _buffer, std::size_t _size);
+    virtual bool encode(bcos::bytes& _buffer) override;
+    // virtual int64_t decode(const bcos::byte* _buffer, std::size_t _size);
+    virtual ssize_t decode(bytesConstRef _buffer) override;
 
 private:
     uint16_t m_type{0};
     uint16_t m_status{0};
-    std::shared_ptr<bcos::bytes> m_seq;
+    std::string m_seq;
```
Comment (ywy2090) [inline bcos-boostssl/websocket/WsMessage.h]: m_seq(SEQ_LENGTH, '0')

File: bcos-boostssl/websocket/WsMessage.h (hunk 2 above)
Comment (LucasLi1024) [reply bcos-boostssl/websocket/WsMessage.h]: fixed

File: bcos-boostssl/websocket/WsMessage.cpp (hunk 1 above)
Comment (LucasLi1024) [reply bcos-boostssl/websocket/WsMessage.cpp]: fixed

File: bcos-boostssl/websocket/WsMessage.h (hunk 3 above)
Comment (LucasLi1024) [reply bcos-boostssl/websocket/WsMessage.h]: fixed

**Pull request merged**